jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -1890,8 +1857,8 @@ Method getMethodHelper(
 			}
 		}
 	}
-	if (!searchAllMethods && initialResultShouldBeReplaced) {
-		// The initial result doesn't match the searching method type.
+	if (initialResultShouldBeReplaced) {
+		// The initial result is not a public method to be searched.
```

Comment:
I don't think this comment is accurate; `initialResultShouldBeReplaced` is modified on line 1849.

Reply:
Updated the comment.